### DISCUSSION TITLE
Overlapping patches and Sliding Window mechanism

### DISCUSSION THREAD

**Discussion by Aayush2003Gupta** [mlcommons/GaNDLF#670] [Q&A]
Hi!!
I had some questions regarding GaNDLF. I would be really grateful if I could get some help regarding them.

Does GaNDLF support `overlapping patches` or the `sliding window mechanism` during inference/training ?

Thanks a lot in advance!!!
Aayush Gupta

**Comment by sarthakpati** [comment]
Hi,

GaNDLF _kind of_ supports this.

- For histology images [[ref](https://mlcommons.github.io/GaNDLF/usage/#offline-patch-extraction-for-histology-images-only)]: You can use the configuration parameters `overlap_factor` and `read_type: sequential`.
- For radiology images: This is a bit more complicated, and it is not possible (at least right now) to perform sliding window mechanism for patch extraction. You can use the queue parameters [[ref](https://mlcommons.github.io/GaNDLF/customize/#:~:text=memory%20as%20tensors-,Queue%20configuration,-%3A%20this%20defines%20how)] to kind of simulate this. Specifically, you can use `q_samples_per_volume`. Let's say your patch size for training is `[128,128,128]` and your total image size per modality is `[240,240,155]` with `patch_sampler: uniform`, you can use `q_samples_per_volume: 5` to ensure the entire volume is captured (there will be overlap which you , though).

Hope this helps!
